src/tiered/tiered_config.c

```diff
@@ -0,0 +1,220 @@
+        if (WT_STRING_MATCH(bstorage->bucket, bucket.str, bucket.len) &&
+          (WT_STRING_MATCH(bstorage->bucket_prefix, prefix.str, prefix.len))) {
+            *bstoragep = bstorage;
+            goto out;
+        }
+    }
+
+    WT_ERR(__wt_calloc_one(session, &new));
+    WT_ERR(__wt_strndup(session, bucket.str, bucket.len, &new->bucket));
+    WT_ERR(__wt_strndup(session, prefix.str, prefix.len, &new->bucket_prefix));
+    storage = nstorage->storage_source;
+#if 0
+    if (storage->customize != NULL) {
+        custom = NULL;
+        WT_ERR(storage->customize(storage, &session->iface, cfg_arg, &custom));
+        if (custom != NULL) {
+            bstorage->owned = 1;
+            storage = custom;
+        }
+    }
+#endif
+    new->storage_source = storage;
+    /* If we're creating a new bucket storage, parse the other settings into it.  */
+    TAILQ_INSERT_HEAD(&nstorage->bucketqh, new, q);
+    TAILQ_INSERT_HEAD(&nstorage->buckethashqh[hash_bucket], new, hashq);
+    F_SET(new, WT_BUCKET_FREE);
+    WT_ERR(__tiered_common_config(session, cfg, new));
+    *bstoragep = new;
+
+out:
+    __wt_spin_unlock(session, &conn->storage_lock);
+    return (0);
+
+err:
+    __wt_spin_unlock(session, &conn->storage_lock);
+    return (ret);
+}
+
+/*
+ * __wt_tiered_conn_config --
+ *     Parse and setup the storage server options for the connection.
+ */
+int
+__wt_tiered_conn_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfig)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+
+    conn = S2C(session);
+
+    if (!reconfig)
+        WT_RET(__wt_tiered_bucket_config(session, cfg, &conn->bstorage));
+
+    /* If the connection is not set up for tiered storage there is nothing more to do. */
+    if (conn->bstorage == NULL)
+        return (0);
+    __wt_verbose(session, WT_VERB_TIERED, "TIERED_CONFIG: bucket %s", conn->bstorage->bucket);
+    __wt_verbose(
+      session, WT_VERB_TIERED, "TIERED_CONFIG: prefix %s", conn->bstorage->bucket_prefix);
+
+    /*
+     * If reconfiguring, see if the other settings have changed on the system bucket storage.
+     */
+    WT_ASSERT(session, conn->bstorage != NULL);
+    if (reconfig)
+        WT_ERR(__tiered_common_config(session, cfg, conn->bstorage));
+
+    WT_STAT_CONN_SET(session, tiered_object_size, conn->bstorage->object_size);
+    WT_STAT_CONN_SET(session, tiered_retention, conn->bstorage->retain_secs);
+
+    return (0);
+
+err:
+    __wt_free(session, conn->bstorage->auth_token);
+    __wt_free(session, conn->bstorage->bucket);
+    __wt_free(session, conn->bstorage->bucket_prefix);
+    __wt_free(session, conn->bstorage);
+    return (ret);
+}
+
+/*
+ * __wt_tiered_config_bstorage --
+ *     Return a bucket storage handle based on the configuration.
+ */
+int
+__wt_tiered_config_bstorage(
+  WT_SESSION_IMPL *session, const char **cfg, WT_BUCKET_STORAGE **bstoragep)
+{
+    WT_BUCKET_STORAGE *bstorage;
```

Comment:
Unused variable? (It's set at the end of the function, but never used.)

Reply:
Good catch! Thanks. There used to be more there that got moved/changed... Fixed.